src/web/client/common/localStore.ts

```diff
@@ -88,16 +116,32 @@ export async function websiteIdtoLanguageMap(accessToken: string, dataverseOrg:
         }
         else {
             showErrorDialog("Error processing the adx_websitelanguages response", "WebsiteLanguage code response failure");
-            throw e;
         }
     }
-    return { websiteIdtoLanguage };
+    return websiteIdtoLanguage;
 }
 
-export async function setContext(accessToken: any, dataverseOrg: any) {
+
+export async function setContext(accessToken: any, pathEntity: string, entityId: string, queryParamsMap: any, portalsFS: PortalsFS) {
+    const entity = pathparam_schemaMap.get(pathEntity) as string;
     dataSourcePropertiesMap = getDataSourcePropertiesMap();
-    ({ websiteIdtoLanguage } = await websiteIdtoLanguageMap(accessToken, dataverseOrg, WEBSITE_LANGUAGES));
-    ({ languageIdCodeMap } = await languageIdtoCodeMap(accessToken, dataverseOrg, PORTAL_LANGUAGES));
+    entitiesSchemaMap = getEntitiesSchemaMap();
+    websiteIdtoLanguage = await websiteIdtoLanguageMap(accessToken, queryParamsMap.get('orgUrl'), WEBSITEID_LANGUAGE);
+    websitelanguageIdtoportalLanguageMap = await websitelanguageIdtoportalLanguage(accessToken, queryParamsMap.get('orgUrl'), WEBSITE_LANGUAGES);
+    languageIdCodeMap = await languageIdtoCode(accessToken, queryParamsMap.get('orgUrl'), PORTAL_LANGUAGES);
```

Comment:
rather than fetching orUrl from the map multiple times, store it in a local variable.